torch/testing/_internal/common_methods_invocations.py

```diff
@@ -1362,6 +1362,22 @@ def _make_tensor_helper(shape, low=None, high=None):
 
     return samples
 
+
+def sample_inputs_cumsum(op_info, device, dtype, requires_grad):
+    def _make_tensor_helper(shape, low=None, high=None):
+        return make_tensor(shape, device, dtype, low=low, high=high, requires_grad=requires_grad)
+
+    samples = (
+        SampleInput((_make_tensor_helper((S, S, S)), 0)),
+        SampleInput((_make_tensor_helper((S, S, S)), 1)),
+        # FIXME : inplace grad fails for the following sample due to mismatch in
```

Comment:
does it only fail for complex inputs or both real and complex inputs?

Reply:
Changing it to
```
kwarg_dtype = torch.cdouble if dtype.is_complex else torch.double
SampleInput((_make_tensor_helper((S, S, S)), 1), kwargs={'dtype': kwarg_dtype}),
```

works as we do gradcheck with torch.double and torch.cdouble but 
`test_variant_consistency_eager_cumsum` fails for `float16, float32, cfloat` with type mismatch in the inplace variant check.